From c731dc2cc4c49e11d6bb7f5c22c1b2e5aef78f0d Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Mon, 17 Nov 2014 16:51:10 +0800
Subject: [PATCH] MLK-9428 dts: imx6sx: optimize can pad settings

Detailed reproduce steps:
1. boot-up to Linux command prompt.
2. send data from CAN device using "candump" command.
3. capture the TXD waveform during transmission.
4. severe overshoot/undershoot is observed (+4.4V ~ -1.2V).

HW team found that the pad setting of the CAN signal pins is not optimized.
In existing BSP, SPEED/DSE/SRE = 10/110/1 is used. They propose change it
to SPEED/DSE/SRE = 00/100/0.

Conflicts:

	arch/arm/boot/dts/imx6sx.dtsi

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit 0e0d95bed6937f4846aeda27de4e93bb141be492)
(cherry picked from commit 2914d58e0182795727930fa9f0bd4236ab9ded3c)
---
 arch/arm/boot/dts/imx6sx-sdb.dts | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index eb564cd..8a5a9ef 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -787,15 +787,15 @@
 
 		pinctrl_flexcan1: flexcan1grp {
 			fsl,pins = <
-				MX6SX_PAD_QSPI1B_DQS__CAN1_TX		0x1b0b0
-				MX6SX_PAD_QSPI1A_SS1_B__CAN1_RX		0x1b0b0
+				MX6SX_PAD_QSPI1B_DQS__CAN1_TX		0x1b020
+				MX6SX_PAD_QSPI1A_SS1_B__CAN1_RX		0x1b020
 			>;
 		};
 
 		pinctrl_flexcan2: flexcan2grp {
 			fsl,pins = <
-				MX6SX_PAD_QSPI1B_SS1_B__CAN2_RX		0x1b0b0
-				MX6SX_PAD_QSPI1A_DQS__CAN2_TX		0x1b0b0
+				MX6SX_PAD_QSPI1B_SS1_B__CAN2_RX		0x1b020
+				MX6SX_PAD_QSPI1A_DQS__CAN2_TX		0x1b020
 			>;
 		};
 
-- 
1.8.0

